src/ast/Event.ts

```diff
@@ -210,9 +209,36 @@ export default class Event extends Node {
       }
     });
 
-    // typecheck guest
-    this.guests.forEach(guest => {});
+    // typecheck guests
+    if (this.guests.length) {
+      this.guests.forEach(guest => {
+        const guestMapValue = map[guest];
+
+        if (!(guestMapValue[0] === "Guest")) {
+          throw new TypeCheckError(
+            {
+              expected: "a valid Guest",
+              actual: guestMapValue[0]
```

Comment:
Do you think it's worth adding the variable name to the error message, since there could be multiple variables on a line?